src/pty.rs

```diff
@@ -35,7 +34,9 @@ impl Pty {
     /// Returns an error if the device node to open could not be determined,
     /// or if the device node could not be opened.
     pub fn pts(&self) -> crate::Result<Pts> {
-        Ok(Pts(self.0.get_ref().pts()?))
+        let pts = Pts(self.0.get_ref().pts()?);
+        self.0.get_ref().set_nonblocking()?;
+        Ok(pts)
```

Comment:
So we do it here. And we're not certain whether this is idempotent enough to not be an issue with multiple calls.